dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -1634,6 +1635,8 @@ public Response addFileToDataset(@PathParam("id") String idSupplied,
 
         try {
             datasetsValidators.validateFileTermsOfUseDTO(optionalFileParams.getFileTermsOfUseDTO());
+        } catch (MissingArgumentException ex) {
```

Comment:
Only suggestion:
It should be possible to merge this and following case in one catch:
```java
catch (MissingArgumentException | InvalidParameterException ex) {
    return error(Response.Status.BAD_REQUEST, ex.getMessage());
} catch (…)
```